core/eolearn/core/utils/parsing.py

```diff
@@ -13,15 +13,32 @@
 """
 from __future__ import annotations
 
+import sys
 from itertools import repeat
-from typing import TYPE_CHECKING, Iterable, List, Optional, Sequence, Tuple, Union, cast
+from typing import TYPE_CHECKING, Dict, Iterable, List, Optional, Sequence, Tuple, Union, cast
 
 from ..constants import FeatureType
+from .types import EllipsisType
+
+if sys.version_info < (3, 8):
+    from typing_extensions import Literal
+else:
+    from typing import Literal  # pylint: disable=ungrouped-imports
 
 if TYPE_CHECKING:
     from ..eodata import EOPatch
 
 
+SingleFeatureSpec = Union[
+    Literal[FeatureType.TIMESTAMP, FeatureType.BBOX], Tuple[FeatureType, str], Tuple[FeatureType, str, str]
+]
+SequenceFeatureSpec = Sequence[Union[SingleFeatureSpec, Tuple[FeatureType, EllipsisType]]]
```

Comment:
Could `Tuple[FeatureType, EllipsisType]` not be considered as a `SingleFeatureSpec` ?

Reply:
Most of the time it represents multiple features, since `(FeatureType.DATA, ...)` gets you all the data features. These cases are really hard to classify...

Reply:
I guess this is also one of those cases where it ended up hurting us that a dict-like functionality is possible in the first place. I don't think it's such a big issue to just provide a list of features, so perhaps this could as well be ultimately deprecated.

Reply:
Nothing left to do here, perhaps just write down these ideas somewhere so it doesn't get lost.

Resolving.